Sources/Sentry/SentryNSProcessInfoWrapper.mm

```diff
@@ -1,6 +1,26 @@
 #import "SentryNSProcessInfoWrapper.h"
 
-@implementation SentryNSProcessInfoWrapper
+@implementation SentryNSProcessInfoWrapper {
+#if TEST
+    NSString *_executablePath;
+}
+- (void)setProcessPath:(NSString *)path
+{
+    _executablePath = path;
+}
+#    define EXECUTABLE_PATH _executablePath;
```

Comment:
Please always prefix macro names with `SENTRY_`. While this instance won't break compilation, this name collides with a build setting, so it's easy to imagine it causing confusion later: 
![image](https://github.com/getsentry/sentry-cocoa/assets/3241469/38a448ce-e46a-4a6a-9be9-8a314aa07031)